File: app/src/main/java/com/trevorwiebe/trackacow/data/db/dao/UserDao.java
package com.trevorwiebe.trackacow.data.db.dao;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

import com.trevorwiebe.trackacow.data.db.entities.UserEntity;

import java.util.List;

@Dao
public interface UserDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertUser(UserEntity userEntity);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertUserList(List<UserEntity> userEntities);

    @Query("SELECT * FROM user WHERE uid = :uid")
    UserEntity getUserByUid(String uid);

    @Update
    void updateUser(UserEntity userEntity);

    @Delete
    void deleteUser(UserEntity userEntity);

    @Query("DELETE FROM user")
    void deleteUserTable();

}

File: app/src/main/java/com/trevorwiebe/trackacow/data/local/dao/UserDao.kt
package com.trevorwiebe.trackacow.data.local.dao

import androidx.room.*
import com.trevorwiebe.trackacow.data.entities.UserEntity

@Dao
interface UserDao {

    @Deprecated("use suspend function")
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertUser(userEntity: UserEntity?)

    @Deprecated("use suspend function")
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertUserList(userEntities: List<UserEntity?>?)

    @Deprecated("use flow function")
    @Query("SELECT * FROM user WHERE uid = :uid")
    fun getUserByUid(uid: String?): UserEntity?

    @Deprecated("use suspend function")
    @Update
    fun updateUser(userEntity: UserEntity?)

    @Deprecated("use suspend function")
    @Delete
    fun deleteUser(userEntity: UserEntity?)

    @Deprecated("use suspend function")
    @Query("DELETE FROM user")
    fun deleteUserTable()
}